### PR TITLE
itests: Test FEVM recursive calls

### PR DESCRIPTION
**Pull request by magik6k**
## Related Issues
On top of https://github.com/filecoin-project/lotus/pull/10010
Closes https://github.com/filecoin-project/ref-fvm/issues/1322

## Proposed Changes
Add 2 itests:
* One with a function which calls itself recursively
* One with a function which calls itself recursively, then calls the contract recursively

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: itests/fevm_test.go [hunk 1]
```diff
@@ -112,3 +95,109 @@ func TestFEVMETH0(t *testing.T) {
...
+	t.Run("n=200,r=10", testN(200, 10))
+	t.Run("n=251,r=10", testN(251, 10))
+
+	t.Run("n=0,r=32", testN(0, 32))
+	t.Run("n=1,r=32", testN(1, 32))
+	t.Run("n=20,r=32", testN(20, 32))
+	t.Run("n=200,r=32", testN(200, 32))
+	t.Run("n=251,r=32", testN(251, 32))
+
+	t.Run("n=0,r=33", testN(0, 33))
```
Comment (magik6k) [inline itests/fevm_test.go:202]: The tests above (with r <= 32) all pass, at 32 (those are recursive actor calls) I'm seeing
```
    evm.go:237: gas used: 102988605
    evm.go:238: 
        	Error Trace:	/home/magik6k/github.com/filecoin-project/go-lotus/itests/evm.go:238
        	            				/home/magik6k/github.com/filecoin-project/go-lotus/itests/fevm_test.go:179
        	Error:      	Should be true
        	Test:       	TestFEVMRecursiveActorCall
        	Messages:   	contract execution failed: 33
```

Which I'm not exactly sure what that means. I'd expect the actor invocation limit to be 1k. Did that change recently?

Comment (Stebalien) [reply itests/fevm_test.go:202]: It should be 1k, but we might be hitting a memory limit here?

Comment (magik6k) [reply itests/fevm_test.go]: Turns out this was https://github.com/filecoin-project/lotus/issues/10041

File: itests/fevm_test.go [hunk 2]
```diff
@@ -139,3 +141,78 @@ func TestEVMRpcDisable(t *testing.T) {
...
+			result := client.EVM().InvokeContractByFuncName(ctx, fromAddr, actorAddr, "exec1(uint256)", inputData)
+			require.Equal(t, result, []byte{})
+		}
+	}
+
+	t.Run("n=0", testN(0))
+	t.Run("n=1", testN(1))
+	t.Run("n=20", testN(20))
+	t.Run("n=200", testN(200))
+	t.Run("n=293", testN(293))
```
Comment (travisperson) [inline itests/fevm_test.go:168]: Is anything special about 293? Would there be any benefit to also testing to exit code 23 in this test?

**Review by travisperson — APPROVED** [review]

**Pull request merged**